packages/node/src/builders/compile/compile.impl.ts

```diff
@@ -0,0 +1,158 @@
+import * as treeKill from 'tree-kill';
+
+export interface NodeCompilerBuilderOptions extends JsonObject {
+  main: string;
+  tsConfig: string;
+  outputPath: string;
+  watch: boolean;
+  sourceMap: boolean;
+  assets: Array<AssetGlob | string>;
+  packageJson: string;
+}
+
+type AssetGlob = {
+  glob: string;
+  input: string;
+  ignore: string[];
+  output: string;
+};
+
+export default createBuilder(run);
+
+function run(options: NodeCompilerBuilderOptions, context: BuilderContext) {
+  return compileTypeScriptFiles(options, context).pipe(
+    tap(() => {
+      updatePackageJson(options, context);
+      copyAssetFiles(options, context);
+    })
+  );
+}
+
+let tscProcess: ChildProcess;
+function compileTypeScriptFiles(
+  options: NodeCompilerBuilderOptions,
+  context: BuilderContext
+): Observable<BuilderOutput> {
+  if (tscProcess) {
+    killProcess(context);
+  }
+  // Cleaning the /dist folder
+  removeSync(join(context.workspaceRoot, options.outputPath));
+
+  return Observable.create((subscriber: Subscriber<BuilderOutput>) => {
+    try {
+      let args = [
+        '-p',
+        join(context.workspaceRoot, options.tsConfig),
+        '--outDir',
+        join(context.workspaceRoot, options.outputPath)
+      ];
+
+      if (options.sourceMap) {
+        args.push('--sourceMap');
+      }
+
+      const tscPath = join(
+        context.workspaceRoot,
+        '/node_modules/typescript/bin/tsc'
+      );
+      if (options.watch) {
+        args.push('--watch');
+        tscProcess = fork(tscPath, args, { stdio: [0, 1, 2, 'ipc'] });
+        subscriber.next({ success: true });
+      } else {
+        tscProcess = fork(tscPath, args, { stdio: [0, 1, 2, 'ipc'] });
+        tscProcess.on('exit', code => {
+          code === 0
+            ? subscriber.next({ success: true })
+            : subscriber.error('Could not compile Typescript files');
+          subscriber.complete();
+        });
+      }
+    } catch (error) {
+      if (tscProcess) {
+        killProcess(context);
+      }
+      subscriber.error(
+        new Error(`Could not compile Typescript files: \n ${error}`)
+      );
+    }
+  });
+}
+
+function killProcess(context: BuilderContext): void {
+  return treeKill(tscProcess.pid, 'SIGTERM', error => {
+    tscProcess = null;
+    if (error) {
+      if (Array.isArray(error) && error[0] && error[2]) {
+        const errorMessage = error[2];
+        context.logger.error(errorMessage);
+      } else if (error.message) {
+        context.logger.error(error.message);
+      }
+    }
+  });
+}
+
+function updatePackageJson(
+  options: NodeCompilerBuilderOptions,
+  context: BuilderContext
+) {
+  const mainFile = relative(dirname(options.main), options.main);
+  const typingsFile = mainFile.replace('.ts', '.d.ts');
+  const mainJsFile = mainFile.replace('.ts', '.js');
+  const packageJson = readJsonFile(
+    join(context.workspaceRoot, options.packageJson)
+  );
+  packageJson.main = `./${mainJsFile}`;
```

Comment:
Consider when someone changes where tsconfig.json root is.. We.shouldn't assume people won't do weird stuff :)